### DISCUSSION TITLE
Slew margin in openlane flow

### DISCUSSION THREAD

**Discussion by akhilesh911** [The-OpenROAD-Project/OpenLane#1919] [Q&A] [answered]
Good morning everyone,
After completion of openlane flow, I noticed the rcx_sta.slew.rpt. It shows the slew limit 4.00 for every design. 

 report_check_types -max_slew -max_cap -max_fanout -violators

max slew

Pin                                           Limit    Slew   Slack

_48482_/A1                              4.00    8.01   -4.01 (VIOLATED)
_54806_/Q                               4.00    8.01   -4.01 (VIOLATED)
hold92/Z                                   4.00    5.89   -1.89 (VIOLATED)
_48476_/A1                              4.00    5.75   -1.75 (VIOLATED)
_54803_/Q                               4.00    5.75   -1.75 (VIOLATED)
_42824_/A3                              4.00    5.72   -1.72 (VIOLATED)
_42823_/ZN                              4.00    5.72   -1.72 (VIOLATED)
_48478_/A1                              4.00    5.34   -1.34 (VIOLATED)

Is there any way to increase or decrease that limit?
Kindly help.

**Comment by vijayank88** [comment]
`set ::env(GLB_RESIZER_MAX_SLEW_MARGIN) 10` check default variable. you can increase or decrease as per need.

**Reply by akhilesh911** [reply]
Yes I tried by changing that value and also placement slew margin but still it's limit is 4.

**Reply by vijayank88** [reply]
@akhilesh911 is this built-in example or own design?
If own design mean need source files to check the issue.

**Reply by akhilesh911** [reply]
It is own design. The slew margin is dependent on clock period, isn't it? My clock period is 50.

**Reply by kareefardi** [reply]
`GLB_RESIZER_MAX_SLEW_MARGIN` has nothing to do with adjusting the constraint.

**Comment by kareefardi** [comment]
Can you share your config? Slew margin used to be a percentage of the clock period.

**Reply by kareefardi** [reply]
and the OpenLane version that you are using

**Reply by akhilesh911** [reply]
[config (copy).txt](https://github.com/The-OpenROAD-Project/OpenLane/files/12213998/config.copy.txt)
This is my config file.

I am using the version of OpenLane 90d369bcce5cf1133c28291bc3c8b6473a19ff62

**Reply by vijayank88** [reply]
@akhilesh911 you're using last year December build.
Try to update to latest build and try again.

**Reply by kareefardi** [reply, marked as answer]
You are using `gf180`. The max transition set for pins in the PDK lib files is `4.0`. If you want to overwrite it, you can update to latest OpenLane and use `MAX_TRANSITION_CONSTRAINT`. You can also provide a custom sdc file and keep using your old version of OpenLane

**Reply by akhilesh911** [reply]
I understood. Thank you @kareefardi  @vijayank88
